.github/workflows/release.yml

```diff
@@ -8,7 +8,7 @@ jobs:
   build:
     name: Build distribution
     runs-on: ubuntu-latest
-
+    if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/main'
```

Comment:
to ensure that the build runs when either `publish-to-pypi`  or `publish-to-testpypi` need it